### DISCUSSION TITLE
How to convert ObjectPath to std::string?

### DISCUSSION THREAD

**Discussion by ttmms** [Kistler-Group/sdbus-cpp#290] [Q&A]
Why is ObjectPath a separate object, but createProxy takes object path argument as std::string and doesn't accept ObjectPath? How do I do conversions in between?

**Comment by oenzan** [comment]
```
    class ObjectPath : public std::string
    {
    public:
      ...
```
As you can see it is already string you don't need to convert.
```
        ObjectPath(std::string path)
            : std::string(std::move(path))
        {}
```
As you can see there is no explicit constructor for string. it can allowed to make implicit conversion.

**Comment by sangelovic** [comment]
@ttmms Originally, since object path is a string, the APIs were designed with `std::string` representation for object paths, interface names, bus names, etc. `ObjectPath` type was introduced in the sdbus-c++ type system to represent object paths as values inside D-Bus messages, to get all the automatic serialization, D-Bus signature etc. by sdbus-c++. There already was an idea of representing both concepts with one type -- `ObjectPath`, but was not deemed too important and was not addressed yet.
